.github/workflows/code_quality.yaml

```
@@ -0,0 +1,60 @@
+# Workflow to run code-quality checks on the source
+name: code-quality
+
+# When will this workflow run?
+on:
+  # run on every push to main, i.e. PR merges
+  push:
+    branches: [ "main" ]
+
+  # Triggers the workflow on pushes to open pull requests with code changes
+  pull_request:
+    paths:
+      - '**.py'
+      - '.github/workflows/code_quality.yaml'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# No permissions required to run this code
+permissions: {}
+
+jobs:
+  code-quality:
+    name: Run code quality checks
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    steps:
+    # Checks-out repository under $GITHUB_WORKSPACE, so job can access it
+    - name: Checkout code
+      with:
+        persist-credentials: true
```

@jwallwork23 cool stuff, I _think_ you can also set credentials to not persist of going public (just check though).
```suggestion
        persist-credentials: false
```

---

Ah yes, I remember this now from zizmor recommendations.